spotbugs/src/main/java/edu/umd/cs/findbugs/detect/InefficientReplaceAll.java

```diff
@@ -0,0 +1,143 @@
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+package edu.umd.cs.findbugs.detect;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.bcel.Const;
+import org.apache.bcel.classfile.Code;
+import org.apache.bcel.classfile.Method;
+
+import edu.umd.cs.findbugs.BugAccumulator;
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.StatelessDetector;
+import edu.umd.cs.findbugs.SystemProperties;
+import edu.umd.cs.findbugs.ba.ClassContext;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import edu.umd.cs.findbugs.classfile.MethodDescriptor;
+
+/**
+ * Find occurrences of replaceAll(String regex, String replacement) without any special regex characters. 
+ *
+ * @author Lasse Lindqvist
+ */
+public class InefficientReplaceAll extends OpcodeStackDetector implements StatelessDetector {
+    private static final boolean DEBUG = SystemProperties.getBoolean("ira.debug");
+
+    private static final List<MethodDescriptor> methods = Collections.singletonList(new MethodDescriptor("", "replaceAll",
+            "(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;"));
+
+    private final BugAccumulator bugAccumulator;
+    
+    private static List<String> regexCharList = new ArrayList<>();
+    static {
+        regexCharList.add(".");
+        regexCharList.add("\\");
+        regexCharList.add("[");
+        regexCharList.add("]");
+        regexCharList.add("{");
+        regexCharList.add("}");
+        regexCharList.add("(");
+        regexCharList.add(")");
+        regexCharList.add("<");
+        regexCharList.add(">");
+        regexCharList.add("*");
+        regexCharList.add("+");
+        regexCharList.add("-");
+        regexCharList.add("=");
+        regexCharList.add("?");
+        regexCharList.add("^");
+        regexCharList.add("$");
+        regexCharList.add("|");
+    }
+
+    public InefficientReplaceAll(BugReporter bugReporter) {
+        this.bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        if (hasInterestingMethod(classContext.getJavaClass().getConstantPool(), methods)) {
+            classContext.getJavaClass().accept(this);
+        }
+    }
+
+    @Override
+    public void visit(Method obj) {
+        if (DEBUG) {
+            System.out.println("------------------- Analyzing " + obj.getName() + " ----------------");
+        }
+        super.visit(obj);
+    }
+
+    @Override
+    public void visit(Code obj) {
+        super.visit(obj);
+        bugAccumulator.reportAccumulatedBugs();
+
+    }
+
+    @Override
+    public void sawOpcode(int seen) {
+        if (DEBUG) {
+            System.out.println("Opcode: " + Const.getOpcodeName(seen));
```

Comment:
Can this use a logger? `java.util.logging.Logger` for now, and we'll convert it later to SLF4J (#635)